src/hotspot/share/cds/dumpTimeClassInfo.hpp

```diff
@@ -209,10 +208,17 @@ class DumpTimeSharedClassTable: public DumpTimeSharedClassTableBaseType
     }
   }
 
-  // Overrides ResourceHashtable<>::iterate(ITER*)
-  template<class ITER> void iterate(ITER* iter) const;
+  template<class ITER> void iterate_all_live_classes(ITER* iter) const;
+  template<typename Function> void iterate_all_live_classes(Function function) const;
+
 private:
-  template<class ITER> class IterationHelper;
+  // It's unsafe to iterate on classes whose loader is dead.
+  // Declare these private and don't implement them. This forces users of
+  // DumpTimeSharedClassTable to use the iterate_all_live_classes() methods
+  // instead.
+  template<class ITER> void iterate(ITER* iter) const;
+  template<typename Function> void iterate(Function function) const;
+  template<typename Function> void iterate_all(Function function) const;
```

Comment:
This helps a lot and thanks for the comment, which describes why there's a wrapper for this class.